dali/ft/filecopy.cpp

```diff
@@ -3771,7 +3774,21 @@ void FileSprayer::updateTargetProperties()
     if (distributedSource)
     {
         if (distributedSource->querySuperFile()==nullptr)
-            distributedSource->addAttrValue("@numDiskReads", totalNumReads);
+        {
+            IPropertyTree & fileAttr = distributedSource->queryAttributes();
+            cost_type readCost = 0;
+            // Check if it is a legacy file without @readCost attribute -> calculate prevReadCost
+            if (!hasReadWriteCostFields(&fileAttr) && fileAttr.hasProp(getDFUQResultFieldName(DFUQRFnumDiskReads)))
+            {
+                stat_type totalReads = fileAttr.getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskReads), 0) + totalNumReads;
+                readCost = money2cost_type(calcFileAccessCost(distributedSource, 0, totalReads));
+            }
+            else
+                readCost = money2cost_type(calcFileAccessCost(distributedSource, 0, totalNumReads));
+
+            distributedSource->addAttrValue(getDFUQResultFieldName(DFUQRFreadCost), readCost);
+            distributedSource->addAttrValue(getDFUQResultFieldName(DFUQRFnumDiskReads), totalNumReads);
```

Comment:
The reason that the distributedTarget update is a set is that I assumed that it is possible to write to a file just once, so the "write" costs shouldn't change after a file is created.  Many reads can occur from a file, so it adds the numReads and readCost in distributedSource.  If I set the values in distributedSource, wouldn't I still need to read the previous cost, add the current cost then update - which is what is happening now? (Similarly, for numDiskReads)

Reply:
As you've mentioned, it is called once after a successful spray operation.  BTW updateTargetProperties has been @numDiskReads and @numDiskWrites for some time.  Without this PR, if updateTargetProperties were to be called multiple times, it would mean the stats would be updated incorrectly.

Reply:
Yes, setting in destination, adding in source makes sense for the reasons you say above. 
It's fine as it is.